node/src/transition_frontier/sync/ledger/snarked/transition_frontier_sync_ledger_snarked_reducer.rs

```diff
@@ -601,6 +611,88 @@ impl TransitionFrontierSyncLedgerSnarkedState {
                     target: target.clone(),
                 };
             }
+            TransitionFrontierSyncLedgerSnarkedAction::P2pDisconnection { peer_id } => {
```

Comment:
I am not very convinced about this one. It was originally part of the `node/p2p` module and the logic made sense there (because it is the upper layer connecting everything), but now this logic is inside the transition frontier module, doesn't feel right. It is also interacting with the snark pool candidate, which has no place here. 

I think it is better to move it with the other callbacks.